impl/src/bazel.rs

```diff
@@ -23,6 +23,35 @@ use crate::{
   util::RazeError,
 };
 
+use std::{
+  env,
+  path::{Path, PathBuf},
+};
+
+/** Returns a path to a Bazel workspace root if one could be found, otherwise None */
+pub fn find_workspace_root(current_dir: Option<&Path>) -> Option<PathBuf> {
+  let default = env::current_dir().unwrap();
+  let cwd: &Path = current_dir.unwrap_or(default.as_path());
+
+  let workspace_files = [
+    cwd.clone().join("WORKSPACE.bazel"),
+    cwd.clone().join("WORKSPACE"),
+  ];
+
+  for workspace in workspace_files.iter() {
+    if workspace.exists() {
+      return Some(cwd.to_path_buf());
+    }
+  }
+
+  match cwd.parent() {
+    None => {
+      return None;
+    }
+    Some(parent) => return find_workspace_root(Some(parent)),
+  };
```

Comment:
A match expression can be the returned from a function, so this could be:

```rust
match cwd.parent() {
  None => None,
  Some(parent) => find_workspace_root(Some(parent)),
}
```
which is just:
```rust
cwd.parent().and_then(|parent| find_workspace_root(Some(parent)))
```

That being said.  It doesn't look like you ever use `find_workspace_root(Some(x))` outside of this recursion.  Can you write this recursion in terms of a while loop?

That would allow you to eliminate the input parameter.

Reply:
(Splitting out an `is_workspace_root(dir: &std::path::Path) -> bool` would probably make that a little more clear)

Reply:
I split out `is_workspace_root`, let me know you're thoughts there.